drv/sprot-api/src/lib.rs

```diff
@@ -83,49 +82,138 @@ pub enum MsgError {
     NonRotError = 18,
     /// A message with version = 0 was received unexpectedly.
     EmptyMessage = 19,
-    /// Update operation failed
-    UpdateError = 20,
 
     /// Insufficient bytes received
-    Incomplete = 21,
+    Incomplete = 20,
     /// Hubpack error
-    Serialization = 22,
+    Serialization = 21,
     /// Sequence number mismatch in Sink test
-    Sequence = 23,
+    Sequence = 22,
+
+    //
+    // Update Related Errors
+    //
+    UpdateBadLength = 23,
+    UpdateInProgress = 24,
+    UpdateOutOfBounds = 25,
+    UpdateTimeout = 26,
+    UpdateEccDoubleErr = 27,
+    UpdateEccSingleErr = 28,
+    UpdateSecureErr = 29,
+    UpdateReadProtErr = 30,
+    UpdateWriteEraseErr = 31,
+    UpdateInconsistencyErr = 32,
+    UpdateStrobeErr = 33,
+    UpdateProgSeqErr = 34,
+    UpdateWriteProtErr = 35,
+    UpdateBadImageType = 36,
+    UpdateAlreadyFinished = 37,
+    UpdateNotStarted = 38,
+    UpdateRunningImage = 39,
+    UpdateFlashError = 40,
+    UpdateSpRotError = 41,
+    UpdateUnknown = 42,
+
     /// Unknown Errors are mapped to 0xff
     Unknown = 0xff,
 }
 
-impl From<u8> for MsgError {
-    fn from(byte: u8) -> MsgError {
+impl From<UpdateError> for SprotError {
+    fn from(value: UpdateError) -> Self {
+        match value {
+            UpdateError::BadLength => SprotError::UpdateBadLength,
+            UpdateError::UpdateInProgress => SprotError::UpdateInProgress,
+            UpdateError::OutOfBounds => SprotError::UpdateOutOfBounds,
+            UpdateError::Timeout => SprotError::UpdateTimeout,
+            UpdateError::EccDoubleErr => SprotError::UpdateEccDoubleErr,
+            UpdateError::EccSingleErr => SprotError::UpdateEccSingleErr,
+            UpdateError::SecureErr => SprotError::UpdateSecureErr,
+            UpdateError::ReadProtErr => SprotError::UpdateReadProtErr,
+            UpdateError::WriteEraseErr => SprotError::UpdateWriteEraseErr,
+            UpdateError::InconsistencyErr => SprotError::UpdateInconsistencyErr,
+            UpdateError::StrobeErr => SprotError::UpdateStrobeErr,
+            UpdateError::ProgSeqErr => SprotError::UpdateProgSeqErr,
+            UpdateError::WriteProtErr => SprotError::UpdateWriteProtErr,
+            UpdateError::BadImageType => SprotError::UpdateBadImageType,
+            UpdateError::UpdateAlreadyFinished => {
+                SprotError::UpdateAlreadyFinished
+            }
+            UpdateError::UpdateNotStarted => SprotError::UpdateNotStarted,
+            UpdateError::RunningImage => SprotError::UpdateRunningImage,
+            UpdateError::FlashError => SprotError::UpdateFlashError,
+            UpdateError::SpRotError => SprotError::UpdateSpRotError,
+            UpdateError::Unknown => SprotError::UpdateUnknown,
+        }
+    }
+}
+
+impl From<u8> for SprotError {
+    fn from(byte: u8) -> SprotError {
```

Comment:
Because we derived `FromPrimitive`, this can simply be
```rust
    fn from(byte: u8) -> SprotError {
        Self::from_u8(byte).unwrap_or(SprotError::Unknown)
    }
```

Reply:
Excellent!

Reply:
Fixed!